### PR TITLE
CD-21: Switch to Gulp

### PR DESCRIPTION
**Pull request by rupl**
This PR adds a Gulp workflow to the repo without removing Grunt. The README has been updated but in order to verify that nothing has gone missing I chose to leave Grunt in place for now.

@left23 this could use your help in order to see how a fresh install works. I've tried my best not to forget anything but nothing beats a second pair of eyes.

Known TODOs:
- **Icon tasks haven't been replaced yet:** we needed to refactor these anyway and remove the dependency on a filesystem location outside the repo
- **Check if deploying to GitHub Pages still works:** I will deploy this to my personal fork's GH Pages to make sure Jekyll still works as expected. Maybe that doesn't need to happen though. ~~@attiks could you could shed some light on how that works, since the repo is absent a `gh-pages` branch? Is the repo just configured to use `master` to deploy to GH Pages? I can't see inside the repo's administrative settings to check this myself.~~ After getting access to admin settings I think this should work. I didn't touch the original Jekyll config at least.
- **Confirm CSS browser support:** I took the liberty of removing the `last 2 versions` keyword from the postcss task since it [creates permanent cruft](https://jamie.build/last-2-versions) and replaced with `>0.333%` instead. It seems like it removed harmless stuff like very old webkit prefixes but I wanted to call it out and make sure. I did that in a separate commit for easy reviewing: ffb0e65

### CODE REVIEW

**Comment by emmajane** [comment]
@rupl fixed permissions for yourself and @left23. now you are both all-powerful. :)

**Comment by left23** [comment]
@rupl This is great, thanks.
Installation works well for me.

Once I run `gulp dev` and make a change to the scss (eg. docs/common-design/sass/cd-footer/_cd-footer.scss), I don't see the changes until I stop and restart the gulp task.

I get a brief notification in the browser from Browsersync for `Sass: docs/styleguide/css/styleguide.scss`

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    .pipe(postcss([
+      prefix({
+        browsers: ['>0.333%', 'iOS 8'],
+        cascade: false,
+      }),
+      // cssnano(),
+    ]))
+    .pipe(gulpif(process.env.NODE_ENV !== 'production', sourcemaps.write('./')))
+    .pipe(gulp.dest(dest))
+    .pipe(gulp.dest('_site/' + dest))
```
Comment (rupl) [inline gulpfile.js]: @left23 the lack of Sass updates til restart probably originates with this line of code, which attempts to dump the compiled CSS into the generated Jekyll site without using Jekyll. I'll look into it.

Comment (left23) [reply gulpfile.js]: Cool. Changes to the html reloads as expected.

**Review by left23 — COMMENTED** [review]
Works as expected. Thanks!

**Pull request merged**